### DISCUSSION TITLE
Empty Upgrades array

### DISCUSSION THREAD

**Discussion by Notadumdum** [miguelgrinberg/python-socketio#1107] [Q&A] [answered]
A client connects to a server... the server then sends back an sid together with an empty upgrades array,ping interval and timeout... does this mean the server doesn't upgrade and still uses polling

**Comment by miguelgrinberg** [comment, marked as answer]
It means the server does not offer upgrades to the client. See the Socket.IO protocol if you want to learn about this: https://github.com/socketio/engine.io-protocol and https://github.com/socketio/socket.io-protocol

**Reply by Notadumdum** [reply]
I've read the documentation multiple times... all basically the same thing.... please give me a specific answer.. is it polling or not

**Reply by miguelgrinberg** [reply]
I can't give you a specific answer, you are asking a generic question.

When a client connects with transport X, an empty upgrades array means that there are no options to upgrade, so the client has to stay with transport X.

If X == polling, then it is polling.
If X == websocket, then it is websocket.

Does this make it more clear?

**Reply by Notadumdum** [reply]
Thanks again
